Stratis.Bitcoin/Wallet/Wallet.cs

```diff
@@ -387,7 +442,20 @@ public HdAddress()
         /// </returns>
         public bool IsChangeAddress()
         {
-            return int.Parse(this.HdPath.Split('/')[4]) == 1;
+            if (string.IsNullOrWhiteSpace(this.HdPath))
+                return false;
+
+            var hdPathParts = this.HdPath.Split('/');
+            if (hdPathParts.Length < 5)            
+                return false;
+
+            int result = 0;
+            if (int.TryParse(hdPathParts[4], out result))
```

Comment:
in this class you're suggesting we do (100% with you on this), we should maybe have something to validate whether an hdpath is formed correctly. 
so it'd have the checks you're doing here: 
- check number of parts (6)
- check int in part 5
etc...